ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/batch/BatchedSubmissionValidator.scala

```diff
@@ -41,13 +39,15 @@ object BatchedSubmissionValidator {
       conflictDetection: ConflictDetection,
       metrics: Metrics,
       ledgerDataExporter: LedgerDataExporter,
+      logEntryIdComputationStrategy: LogEntryIdComputationStrategy,
```

Comment:
This is a simple function of `Raw.Envelope => DamlLogEntryId` so you might as well just inject it as-is instead of using a `trait`.

Reply:
I disagree; a trait makes it way easier to search for implementations later.